src/test/rpc/Simulate_test.cpp

```diff
@@ -0,0 +1,974 @@
+
+        {
+            auto validateOutput = [&](Json::Value const& resp,
+                                      Json::Value const& tx) {
+                auto result = resp[jss::result];
+                checkBasicReturnValidity(
+                    result, tx, 5, env.current()->fees().base * 2);
+
+                BEAST_EXPECTS(
+                    result[jss::engine_result] == "tefBAD_SIGNATURE",
+                    result[jss::engine_result].toStyledString());
+                BEAST_EXPECT(result[jss::engine_result_code] == -186);
+                BEAST_EXPECT(
+                    result[jss::engine_result_message] ==
+                    "A signature is provided for a non-signer.");
+
+                BEAST_EXPECT(
+                    !result.isMember(jss::meta) &&
+                    !result.isMember(jss::meta_blob));
+            };
+
+            Json::Value tx;
+
+            tx[jss::Account] = alice.human();
+            tx[jss::TransactionType] = jss::AccountSet;
+            tx[sfDomain] = newDomain;
+            tx[sfSigners] = Json::arrayValue;
+            {
+                Json::Value signer;
+                signer[jss::Account] = becky.human();
+                signer[jss::SigningPubKey] = strHex(dylan.pk().slice());
+                Json::Value signerOuter;
+                signerOuter[sfSigner] = signer;
+                tx[sfSigners].append(signerOuter);
+            }
+
+            // test with autofill
+            testTx(env, tx, validateOutput, false);
+
+            tx[sfSigningPubKey] = "";
+            tx[sfTxnSignature] = "";
+            tx[sfSequence] = env.seq(alice);
+            // transaction requires a non-base fee
+            tx[sfFee] =
+                (env.current()->fees().base * 2).jsonClipped().asString();
+            tx[sfSigners][0u][sfSigner][jss::TxnSignature] = "";
+
+            // test without autofill
+            testTx(env, tx, validateOutput);
+
+            // TODO: check that the ledger wasn't affected
+        }
+    }
+
+    void
+    testDeleteExpiredCredentials()
+    {
+        testcase("Credentials aren't actually deleted on `tecEXPIRED`");
+
+        // scenario setup
+
+        using namespace jtx;
+        Env env(*this);
+
+        Account const subject{"subject"};
+        Account const issuer{"issuer"};
+
+        env.fund(XRP(10000), subject, issuer);
+        env.close();
+
+        auto const credType = "123ABC";
+
+        auto jv = credentials::create(subject, issuer, credType);
+        uint32_t const t =
+            env.current()->info().parentCloseTime.time_since_epoch().count();
+        jv[sfExpiration.jsonName] = t;
+        env(jv);
+        env.close();
+
+        {
+            auto validateOutput = [&](Json::Value const& resp,
+                                      Json::Value const& tx) {
+                auto result = resp[jss::result];
+                checkBasicReturnValidity(
+                    result, tx, 4, env.current()->fees().base);
```

Comment:
Please change 4 -> seq() value / const

Reply:
fixed in 0bbf589f53a13069448d709faa5cea88b6ee547f